src/track/track.cpp

```diff
@@ -780,14 +780,23 @@ void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
     for (const auto& pCue: m_cuePoints) {
         connect(pCue.get(), &Cue::updated, this, &Track::slotCueUpdated);
         // update main cue point
-        if (pCue->getType() == Cue::Type::MainCue) {
+        if (pCue->getType() == mixxx::CueType::MainCue) {
             m_record.setCuePoint(CuePosition(pCue->getPosition()));
         }
     }
     markDirtyAndUnlock(&lock);
     emit cuesUpdated();
 }
 
+void Track::setCuePoints(const QList<mixxx::CueInfo>& cueInfos) {
+    QList<CuePointer> cuePoints;
```

Comment:
Obtain both track id and the (type-safe) sample rate once at the beginning of this function directly from m_record while the object is locked.

Reply:
Like this?

```c++
TrackId trackId;
mixxx::AudioSignal::SampleRate sampleRate;
{
    QMutexLocker lock(&m_qMutex);
    trackId = m_record.getId();
    sampleRate = getSampleRate();
} // implicitly unlocked when leaving scope

Reply:
Almost, see my comment